### ISSUE TITLE
Independent variable of differential is ignored

### ISSUE THREAD

**Issue by lamorton** [SciML/ModelingToolkit.jl#1065]
It's possible to take a derivative w.r.t. some parameter other than the independent variable, and the ODEProblem construction assumes that the differential is actually w.r.t. the independent variable instead. 
```julia
using ModelingToolkit,DifferentialEquations,Test
@parameters t,w
pars = ModelingToolkit.@parameters((A)) 
vars = ModelingToolkit.@variables((u1(t),u2(t)))
Dt = Differential(t)
Dw = Differential(w)
eqs = [
  Dt(u2) ~ t,
  Dw(u1) ~ A,
]

sys = ModelingToolkit.ODESystem(eqs,t,vars,pars)
tspan = (0.0, 1.0)
u0 = [u1 => 0, u2 => 0]
params = [A => 1]
prob = ODEProblem(sys,u0,tspan,params) 
sol = solve(prob,Rodas5())
```
  
I was working on a PR to throw an error during ODESystem construction if any derivatives had a different independent variable, but maybe this is too strict. Maybe the check should happen at ODEProblem construction instead, in case there are systems that would simplify.  For instance, in the above, expanding `Dw(u1)` should yield zero, because `u1` isn't a function of `w`.  OTOH, maybe this simplification needs to take place _before_ constructing the ODESystem. 

[Edit: mistake in the MWE.]

**Comment by lamorton** [comment]
Do you happen to know where I could find an example of this use-case?

**Comment by lamorton** [comment]
Fixed by #1076 .
